working-with-multiple-clusters.md

```diff
@@ -0,0 +1,196 @@
+
+Each context contains a cluster, namespace, and user information that kubectl uses to make the cluster's API calls. You can find all these configurations in the `~/.kube/config` file. However, configuring this file could be overwhelming if you're starting with Kubernetes, especially if you want to start coding your applications. Most of the time, you don't have to configure the kubeconfig file as most of the Kubernetes installation providers already offer a command to generate that file.
+
+If you're already using Kubernetes, you might give it a look in your `~/.kube/config` file, and you might find out that you already have the configuration necessary to connect to different clusters. In case you want to understand more about Kubernetes contexts, you can [visit the official Kubernetes documentation site](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and learn how to [configure access to multiple clusters using contexts](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). In this post, I'll focus on the different ways you have to use contexts to increase your development productivity when working with more than one cluster.
+
+## Using Kubernetes Contexts
+
+Once you have your kubeconfig file configured with the contexts you'll need, it's time to use them.
+
+The idea behind configuring a Kubernetes context is that once you have configured your access to the different clusters, you can run this command:
+
+```
+$ kubectl config use-context docker-desktop
+Switched to context "docker-desktop."
+$ kubectl config use-context remote-cluster
+Switched to context "remote-cluster".
+```
+
+However, the above commands are still a little bit longer. Although you could create aliases in your workstation, I'd highly recommend you to use [kubectx](https://github.com/ahmetb/kubectx), a tool for switching and managing contexts. Besides changing contexts, you can easily go back to the previous contexts, show the current context, rename or delete a context, and more. So the above commands now will be:
+
+```
+$ kubectx docker-desktop
+Switched to context "docker-desktop".
+$ kubectx remote-cluster
+Switched to context "remote-cluster".
+$ kubectx -
+Switched to context "docker-desktop".
+```
+
+Every subsequence command you run will be making calls to the corresponding Kubernetes cluster.
+
+## Working with Multiple Namespaces
+
+At this point, you're able to connect and work with multiple Kubernetes clusters from your workstation. But remember that a context is also referring to a namespace in Kubernetes. That namespace will be the default one when you don't specify a namespace in a kubectl command. For instance, your `docker-desktop` context has the `default` namespace configured. Then, every time you run `kubectl get pods`, it will return the list of pods from the `default` namespace.
```

Comment:
```suggestion
At this point, you're able to connect and work with multiple Kubernetes clusters from your workstation. But remember that a context is also referring to a namespace in Kubernetes. That namespace will be `default`  when you don't specify a namespace in a kubectl command. For instance, your `docker-desktop` context has the `default` namespace configured. Then, every time you run `kubectl get pods`, it will return the list of pods from the `default` namespace.
```